src/react-queries/index.ts

```diff
@@ -18,6 +31,102 @@ import { thorchainLp } from 'pages/ThorChainLP/queries/queries'
 export const thorchainBlockTimeSeconds = '6.1'
 const thorchainBlockTimeMs = bn(thorchainBlockTimeSeconds).times(1000).toNumber()
 
+const common = createQueryKeys('common', {
+  allowanceCryptoBaseUnit: (
+    assetId: AssetId | undefined,
+    spender: string | undefined,
+    from: string | undefined,
+  ) => ({
+    queryKey: ['allowanceCryptoBaseUnit', assetId, spender, from],
+    queryFn: async () => {
+      if (!assetId) throw new Error('assetId is required')
+      if (!spender) throw new Error('spender is required')
+      if (!from) throw new Error('from address is required')
+
+      if (!isToken(fromAssetId(assetId).assetReference)) return null
+      const supportedEvmChainIds = getSupportedEvmChainIds()
+      if (!supportedEvmChainIds.includes(fromAssetId(assetId).chainId as KnownChainIds)) return null
+
+      const allowanceOnChainCryptoBaseUnit = await getErc20Allowance({
+        address: fromAssetId(assetId).assetReference,
+        spender,
+        from,
+        chainId: fromAssetId(assetId).chainId,
+      })
+
+      return allowanceOnChainCryptoBaseUnit
+    },
+    enabled:
+      assetId &&
+      spender &&
+      from &&
+      getSupportedEvmChainIds().includes(fromAssetId(assetId).chainId as KnownChainIds),
+  }),
+})
+
+const mutations = createMutationKeys('mutations', {
+  approve: ({
+    assetId,
+    spender,
+    amount,
+    wallet,
+    from,
+    accountNumber,
+  }: {
+    assetId: AssetId | undefined
+    spender: string | undefined
+    amount: string | undefined
+    wallet: HDWallet | null
+    from: string | undefined
+    accountNumber: number | undefined
+  }) => ({
+    // note how we don't add the wallet here because of its non-serializable nature
+    mutationKey: ['approve', { assetId, spender, amount, from }],
+    mutationFn: async () => {
+      if (!assetId) throw new Error('assetId is required')
+      if (!spender) throw new Error('spender is required')
+      if (!amount) throw new Error('amount is required')
+      if (!from) throw new Error('from address is required')
+      if (!wallet) throw new Error('wallet is required')
+      if (accountNumber === undefined) throw new Error('accountNumber is required')
```

Comment:
Paranoia and type exhaustiveness - these *should* never get hit because `mutate()` is called programmatically and everything should be sane by then, but if one of these were to be missing, this will catch it